### PR TITLE
fix when sidebar show non-md folders

### PR DESCRIPTION
**Pull request by B2o5T**
fixes https://github.com/shuding/nextra/issues/1315

### CODE REVIEW

File: packages/nextra/__test__/__snapshots__/page-map.test.ts.snap [hunk 1]
```diff
@@ -663,8 +624,19 @@ exports[`Page Process > pageMap ru 1`] = `
...
+              "theme": {
+                "layout": "raw",
+              },
+              "type": "page",
+            },
+            "acknowledgement": "🧩 Acknowledgement",
+            "team": "👥 Team",
+          },
           "kind": "Meta",
+          "locale": "en-US",
```
Comment (B2o5T) [inline packages/nextra/__test__/__snapshots__/page-map.test.ts.snap:639]: I guess it fix even silent bug :))

File: packages/nextra/__test__/page-map.test.ts [hunk 2]
```diff
@@ -54,4 +54,12 @@ describe('Page Process', () => {
...
     expect(gettingStartData.pageMap).toEqual(indexData.pageMap)
   })
+
+  it("should not add to pageMap folders that doesn't contain markdown files", async () => {
+    const { items } = await collectFiles(
+      path.join(PAGES_DIR, 'docs/test-non-md-folders'),
+      ['']
+    )
+    expect(items.length).toBe(0)
+  })
```
Comment (B2o5T) [inline packages/nextra/__test__/page-map.test.ts:64]: small test (:

File: packages/nextra/src/plugin.ts [hunk 3]
```diff
@@ -152,15 +152,11 @@ export async function collectFiles(
...
 
-  const mdxPagesAndFolders: (MdxFile | Folder)[] = []
   const mdxPages: MdxFile[] = []
   const metaLocaleIndexes = new Map<string, number>()
 
   for (let i = 0; i < items.length; i++) {
     const item = items[i]
-    if (item.kind === 'MdxPage' || item.kind === 'Folder') {
-      mdxPagesAndFolders.push(item)
-    }
```
Comment (B2o5T) [inline packages/nextra/src/plugin.ts:163]: collection was updated but never used => removed

**Comment by mattvb91** [comment]
Is there any chance we could add a config option to say "subpath" that the sidebar will load only files based FROM that subpath. For example I want to host my docs under `/docs` but I want the sidebar to START at that level. Right now the sidebar would have `docs` as a collapsible but I want that to be the root.

Probably not relevant for this PR?

**Comment by shuding** [comment]
@mattvb91 you can use `"display": "children"` for that case.

File: packages/nextra/src/plugin.ts [hunk 4]
```diff
@@ -183,15 +179,7 @@ export async function collectFiles(
...
 
-    if (metaIndex === undefined) {
-      // Create a new meta file if it doesn't exist.
-      fileMap[metaPath] = {
-        kind: 'Meta',
-        ...(locale && { locale }),
-        data: Object.fromEntries(defaultMeta)
-      }
-      items.push(fileMap[metaPath])
-    } else {
```
Comment (shuding) [inline packages/nextra/src/plugin.ts:194]: @B2o5T It seems like this change caused `callout` and `tabs` to be lowercased:

CleanShot 2023-01-24 at 14 23 13@2x

https://nextra.site/docs/docs-theme/built-ins

Comment (B2o5T) [reply packages/nextra/src/plugin.ts:194]: Saw it yesterday, will fix it !

**Pull request merged**